src/common/utils/http.ts

```diff
@@ -0,0 +1,33 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { app } from "electron";
+import { container } from "readium-desktop/main/di";
+import { RootState } from "readium-desktop/main/redux/states";
+import { Store } from "redux";
+import * as request from "request";
+import { promisify } from "util";
+
+export type httpResponse = request.Response;
+
+export function httpGet(url: string, options?: request.CoreOptions): Promise<request.Response> {
+
+    const store = container.get("store") as Store<RootState>;
+    const locale = store.getState().i18n.locale;
+    const requestOptions = Object.assign(
+        {},
+        { url },
+        {
+            headers: {
+                "User-Agent": "readium-desktop",
+                "Accept-Language": `${locale}-${locale.toUpperCase},en-US;q=0.7,en;q=0.5`,
+            },
+        },
+        options,
+    );
+    return promisify<request.CoreOptions, request.Response>(request)(requestOptions);
```

Comment:
I think that the original `httpGet` promise-wrapper code for the vanilla `request` API was incorrect because HTTP response status codes were not handled properly (e.g. `404` with a perfectly valid `body`, no `error`). I think using `promisify` perpetuates this.

I propose this fix (note that the `request` lib follows HTTP `3xx` redirects by default, so no need to handle this explicitly):

```javascript
return new Promise((resolve, reject) => {

request({
url,
method: "GET",
encoding: undefined, // note that null means Buffer, undefined means .toString(undefined), which defaults to UTF8
}, (error, response, body) => {
    if (error) {
        reject(error);
        return;
    }
    if (!response) {
        // this should never happen because error is falsy,
        // but we fail-fast to move onto the next response check ...
        reject(new Error(`No HTTP response?! ${url}`));
        return;
    }
    if (response.statusCode < 200 || response.statusCode >= 300) {
        const b = body ? JSON.stringify(body) : "";
        reject(new Error(`HTTP response status code: ${url} => ${response.statusCode} => ${b}`));
        return;
    }
    // response.statusCode >= 200 && response.statusCode < 300
   if (!body) {
        reject(new Error(`HTTP no body?! ${url} => ${response.statusCode}`));
        return;
   }

   resolve(body); // this should be UTF8 string, given the default encoding option
});

});
```

Reply:
Naturally, this means removing `.body` from `opdsFeedData.body`, `lsdStatusBodyResponse.body`, `lcplResponse.body`, etc. in other modified code in this PR.

Reply:
done